tests/unit/src/test/scala/tests/GradleRepositoriesTest.scala

```diff
@@ -0,0 +1,95 @@
+package tests
+
+import scala.meta.internal.builds.GradleBuildTool
+
+import coursierapi.Credentials
+import coursierapi.IvyRepository
+import coursierapi.MavenRepository
+import coursierapi.Repository
+
+class GradleRepositoriesTest extends BaseSuite {
+
+  check(
+    Nil,
+    """|  repositories {
+       |    mavenCentral()
+       |  }
+       |""".stripMargin,
+  )
+
+  check(
+    List(
+      Repository.central(),
+      Repositories.google,
+      Repositories.bintrayIvy("testId"),
+    ),
+    """|  repositories {
+       |    mavenCentral()
+       |    maven {
+       |      url "https://maven.google.com"
+       |    }
+       |    ivy {
+       |      url "https://dl.bintray.com/testId"
+       |      patternLayout {
+       |        artifact "[organisation]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)[revision]/[type]s/[artifact](-[classifier]).[ext]"
+       |      }
+       |    }
+       |  }
+       |""".stripMargin,
+  )
+
+  check(
+    List(
+      Repository.central(),
+      Repositories.google.withCredentials(Repositories.testCredentials()),
+      Repositories
+        .bintrayIvy("testId")
+        .withCredentials(Repositories.testCredentials()),
+    ),
+    """|  repositories {
+       |    mavenCentral()
+       |    maven {
+       |      url "https://maven.google.com"
+       |      credentials {
+       |        username "testUser"
+       |        password "testPassword"
+       |      }
+       |    }
+       |    ivy {
+       |      url "https://dl.bintray.com/testId"
+       |      patternLayout {
+       |        artifact "[organisation]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)[revision]/[type]s/[artifact](-[classifier]).[ext]"
```

Comment:
I wonder if `[scalaVersion]` and `sbt_[sbtVersion]` is something gradle will understand :thinking: 

Would you be able to try and add a custom ivy repository and see if this pattern works? Bintray will not work unfortunately, since it no longer exists. Maybe `ivy2Local` would be good to check out

We might need to get rid of of these two.

Reply:
it's fine if it will delete it if that's optional, thanks for confirming!